### ISSUE TITLE
Question about your thesis!

### ISSUE THREAD

**Issue by bw4sz** [MonoCloud/FirstSem#1]
Hey Andreas,

Sorry for hijacking the issue feature, couldn't find an email address.

I just read your thesis on deep segmentation for 3d point clouds. Its great! I am working on tree detection from lidar + hyperspectral + rgb data (public facing [here](http://www.ecodse.org/)). My plan was to try two approaches for the detection process.

1. RCNN (perhaps [Retinanet](https://github.com/bw4sz/keras-retinanet)) + Pointnet for feature classification after the bounding box prediction. 

2. Multiple 2d views of the voxelized point cloud fed into 2D CNN. The attractive part of this would be that I could directly concatenate the (RGB + Hyperspectral + 2D projection Lidar) into one numpy stack to feed into tensorflow.

My reading of your thesis is that 2d projections into Mask RCNN did not work well. Is this your general feeling as well? 

My other question involves data preprocessing. Its poorly explained in many pointnet papers how they decide on the fixed number of points. For any non-benchmark dataset, the points and density will be highly variable. Pointnet++ talks a bit about farthest point sampling, but it isn't clear to me when that happens (before the data input?). You mentioned disc sampling to create more regular points along a voxelgrid. Did you find this effective? Was this done in PCL or in python (I use [open3d](https://github.com/IntelVCL/Open3D))?

Best,

Ben

**Comment by afoltyn** [comment]
Hey Ben,

I didn't know my thesis was online yet :D. The dataset you are using seems
to be really exciting! To your questions:

a) The problem with the multi-view pipeline was that I had images from
different angles and then had to project about 16 different images with
different instance segmentations back onto a single plant point cloud. The
algorithm for merging all views was the bottleneck of this implementation.
I didn't really work intensively on this approach and implemented it
relatively simply. This could be done much better.
But as you have already written, you wouldn't have to do that in your case,
since you only use bird-view 2D projections to get height information, that
you are concatenating with the RGB+hyperspectral (right?). In this case,
the second approach would certainly produce good results :). Is your
ultimate goal object detection or instance segmentation of the crowns?

b) So for PointNet I used a voxel grid (128^3) to reduce the number of
points (with https://github.com/daavoo/pyntcloud) - btw. thanks for the
link to open3d, it looks really good. I only used Poisson Disc Sampling
before meshing the point clouds in the multi-view pipeline. I used MeshLab
and MeshLab Server to automate it. So if you want to have a quick glance
how it performs you can simply use MeshLab to test it.
For the number of points for PointNet I sticked to the original paper.
After a little trial and error, I also noticed that the performance doesn't
really improve if I train with a lot more points. But during inference you
are not bound to the fixed number of points and you can increase/decrease
the size.

Alternatively to PointNet++ you can have a look at PointCNN (
https://arxiv.org/abs/1801.07791) and
https://liuziwei7.github.io/projects/DGCNN - if you haven't already seen
it.

If you have any questions, you can also write me directly. Email:
andreas.foltyn@gmail.com

Best regards,
Andy


2018-06-19 18:52 GMT+02:00 Ben Weinstein <notifications@github.com>:
